execution-engine/src/native_modules/aes.rs

```diff
@@ -102,10 +102,7 @@ impl AesCounterModeService {
             let block_size = cypher.block_size();
             // For some reason, the length of the output buffer needs to be
             // a multiple of the block size and at least two blocks:
-            let padded_size = std::cmp::max(
-                2 * block_size,
-                (input.len() + block_size - 1) / block_size * block_size,
-            );
+            let padded_size = (input.len() + 2 * block_size - 1) / block_size * block_size;
```

Comment:
Understanding what this does requires the reader know Rust operator precedence pretty well, as well as how it handles division of integers. Can we make it clearer by adding parentheses and a comment about the behavior of integer division?

Reply:
Or at least remove the existing comment that states something quite different!

It seems very bizarre that the output buffer has to be a multiple of TWICE the block size. Do we have any idea why that is?

I wonder whether the original comment and code was correct for some other algorithm or configuration or whether it was always just wrong.